### PR TITLE
Search refactor

### PR DESCRIPTION
**Pull request by bhollis**
This rearranges the search/filter code with a goal towards cleaning it up and extracting the logic for which filters we support and how they work from how the component and all that works. This achieves the following:

1. Fixes https://github.com/DestinyItemManager/DIM/issues/2337 by making the filter functions and config generate per-Destiny-version, and regenerating them when we switch version.
2. Fixes https://github.com/DestinyItemManager/DIM/issues/2273 by implementing our own clear button, which replaces the filter-help button when you have stuff typed.
3. Moves search into its own module, and moves filter help into the search-filter component.
4. Fixes up a few of the searches that didn't apply to D2.
5. Add "power" as an alias for "light" and switch to "power" for some D2 stuff.
5. Add "basepower" filter.
5. Add "is:goldborder".

### CODE REVIEW

**Comment by bhollis** [comment]
I'm also experimenting with making the autocomplete suggestions much easier to trigger - they'll now show you options even if your query matches the middle of a keyword. This may help folks discover searches.

autocomplete

**Comment by bhollis** [comment]
I made another tweak so the automcompleter won't show up when there's nothing interesting...

**Comment by duckmanBR** [comment]
This will help with at least 2 problems.  No need to implement alternate delimiters (#2361) since the user won't need to start with `is:` and will clearly show `not:` as a valid option.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,13 @@
 * Add a new smart loadout to gather reputation items for redemption.
 * Scroll the loadout drawer on mobile.
 * Show character level progression under level 20 for D2.
+* Add basepower: filter and is:goldborder filter.
+* Fix filtering in D1.
+* Add a button to clear the current search.
 * Stacks of three or more rare mods now have a yellow border
+* Add basepower: filter and is:goldborder filter.
+* Fix filtering in D1.
+* Add a button to clear the current search.
```
Comment (joshminnie) [inline docs/CHANGELOG.md]: These 3 lines are repeated twice.

**Comment by duckmanBR** [comment]
I only noticed it now, but the "old" **x** button was beside the cheatsheet **?** but now replaces it.

**Comment by bhollis** [comment]
That’s right! I figured it’d be nice to save some space and call attention to that area. 

-Ben

**Pull request merged**